roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -216,10 +218,51 @@ void sendMsgToAzureIoTHub(const char* msg, IOTHUB_DEVICE_CLIENT_HANDLE deviceHan
     messagecount++;
 }
 
+// Message serialization for sending topic messages via reported properties 
+static char* serializeToJson(std::string topic, std::string message)
+{
+    char* result = NULL;
+
+    JSON_Value* root_value = json_value_init_object();
```

Comment:
A defensive coding practice would be to check that this was allocated before using it. (It may throw)